packages/nextjs/src/index.client.ts

```diff
@@ -1,6 +1,6 @@
 import { configureScope, init as reactInit } from '@sentry/react';
 
-import { MetadataBuilder } from './utils/metadataBuilder';
+import { MetadataBuilder, SDK_NAME } from './utils/metadataBuilder';
```

Comment:
I'm not sure if that `MetadataBuilder` dance is really necessary. It adds a lot of boilerplate to set a single value. Especially that both, client and server are using the same values.
It also mutates its input by calling a method, and that input is then passed to init(?). 

```js
function applySdkMetadata(options: NextjsOptions, packageNames: string[]): NextjsOptions {
  return {
    ...options,
    _metadata: {
      ...options._metadata,
      sdk: {
        name: SDK_NAME,
        version: SDK_VERSION,
        packages: packageNames.map((pkgName: string) => ({
          name: `${PACKAGE_NAME_PREFIX}${pkgName}`,
          version: SDK_VERSION,
        })),
      },
    },
  };
}
```

A single function instead of 30LoC class which will have to be transpiled to functions and prototypes for ES5.

Reply:
or mutation version:

```js
function applySdkMetadata(options: NextjsOptions, packageNames: string[]): void {
  options._metadata = options._metadata || {};
  options._metadata.sdk = {
    name: SDK_NAME,
    version: SDK_VERSION,
    packages: packageNames.map((pkgName: string) => ({
      name: `${PACKAGE_NAME_PREFIX}${pkgName}`,
      version: SDK_VERSION,
    })),
  };
}
```